components/src/core/ToolbarMenu/ToolbarMenu.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { HTMLAttributes, useState, useCallback, useRef, useEffect } from 'react';
+import ArrowDropDown from '@material-ui/icons/ArrowDropDown';
+import clsx from 'clsx';
+import createStyles from '@material-ui/core/styles/createStyles';
+import composeRefs from '@seznam/compose-react-refs';
+import { DrawerNavGroup, NavItem } from '../Drawer';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+import Menu, { MenuProps as standardMenuProps } from '@material-ui/core/Menu';
+import PropTypes from 'prop-types';
+import { Theme } from '@material-ui/core/styles/createMuiTheme';
+import useTheme from '@material-ui/core/styles/useTheme';
+import Typography from '@material-ui/core/Typography';
+
+export type ToolbarMenuClasses = {
+    root?: string;
+    dropdownArrow?: string;
+    icon?: string;
+    label?: string;
+    menuItem?: string;
+};
+
+export type ToolbarMenuCompItem = Omit<NavItem, 'itemID'> & { itemID?: string };
+export type ToolbarMenuCompGroup = {
+    /** The color used for the text */
+    fontColor?: string;
+    /** The color used for icons */
+    iconColor?: string;
+    /** List of navigation items to render */
+    items: ToolbarMenuCompItem[];
+    /** Text to display in the group header */
+    title?: string;
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        root: {
+            cursor: 'pointer',
+            fontSize: '1rem',
+            display: 'flex',
+            alignItems: 'center',
+        },
+        cursorPointer: {
+            cursor: 'pointer',
+        },
+        dropdownArrow: {
+            marginLeft: theme.spacing(0.5),
+        },
+        icon: {
+            marginRight: `${theme.spacing(1)}px`,
+            display: 'inline-flex',
+            fontSize: 'inherit',
+        },
+        label: {
+            textOverflow: 'ellipsis',
+            overflow: 'hidden',
+        },
+        navGroups: {
+            '&:active, &:focus': {
+                outline: 'none',
+            },
+        },
+        rotateDropdownArrow: {
+            transform: 'rotate(180deg)',
+        },
+    })
+);
+
+export type ToolbarMenuProps = HTMLAttributes<HTMLSpanElement> & {
```

Comment:
We should extend all of the TypographyProps here as well and spread them to the root element, so the user can do things like add `className` or `variant` props to change some of the appearance. Refer to the ListItemTag for how we did this there.

Reply:
I just tested out the toolbar menu in the showcase doing this:
![image](https://user-images.githubusercontent.com/29152776/151596175-af4e8297-871a-40de-b4ba-ca9ea4549a5a.png)

And comparing it to the standard toolbar, it looks like this:
![image](https://user-images.githubusercontent.com/29152776/151596233-c730245c-58ed-4551-bf01-b79049e90062.png)


I could get the margin to look correct by using `classes={{root: classes.title}}` — it's just a little more inconvenient that using `className` (either one should work).

The bigger issue is that I cannot change the variant to be `h6` to match the expected style for a title line.